kong/plugins/zipkin/tracing_headers.lua

```diff
@@ -314,6 +315,41 @@ local function parse_jaeger_trace_context_headers(jaeger_header)
 end
 
 
+local function parse_datadog_headers(headers)
+  -- allow testing to spy on this.
+  local warn = kong.log.warn
+
+  local should_sample = headers["x-datadog-sampling-priority"]
+  if should_sample == "1" or should_sample == "2" then 
+    should_sample = true
+  elseif should_sample == "0" or should_sample == "-1" then
+    should_sample = false
+  elseif should_sample ~= nil then
+    warn("x-datadog-sampling-priority header invalid; ignoring.")
+  end
+
+  local had_invalid_id = false
+
+  local trace_id = headers["x-datadog-trace-id"]
+  if trace_id and not trace_id:match("%d") then
+    warn("x-datadog-trace-id header invalid; ignoring.")
+    had_invalid_id = true
+  end
+
+  local parent_id = headers["x-datadog-parent-id"]
+  if parent_id and not parent_id:match("%d") then
+    warn("x-datadog-parent-id header invalid; ignoring.")
+    had_invalid_id = true
+  end
+
+  if had_invalid_id then
+    return nil, nil, should_sample
+  end
```

Comment:
I don't understand why had_invalid_id is needed here. Is it only to have 2 warnings in case both the traceid and the parent id are present but not integers? Seems like on that case the traceid warning alone would do just fine.

```suggestion
  local trace_id = headers["x-datadog-trace-id"]
  if trace_id and not trace_id:match("%d") then
    warn("x-datadog-trace-id header invalid; ignoring.")
    return nil, nil, should_sample
  end

  local parent_id = headers["x-datadog-parent-id"]
  if parent_id and not parent_id:match("%d") then
    warn("x-datadog-parent-id header invalid; ignoring.")
    return nil, nil, should_sample
  end
```